homeassistant/components/intesishome/climate.py

```diff
@@ -0,0 +1,413 @@
+        self._max_temp = self._controller.get_max_setpoint(self._deviceid)
+        self._rssi = self._controller.get_rssi(self._deviceid)
+        self._run_hours = self._controller.get_run_hours(self._deviceid)
+        self._target_temp = self._controller.get_setpoint(self._deviceid)
+        mode = self._controller.get_mode(self._deviceid)
+
+        # Operation mode
+        self._hvac_mode = MAP_OPERATION_MODE.get(mode, STATE_UNKNOWN)
+
+        # Swing mode
+        # Climate module only supports one swing setting.
+        if self._has_swing_control:
+            self._vvane = self._controller.get_vertical_swing(self._deviceid)
+            self._hvane = self._controller.get_horizontal_swing(self._deviceid)
+
+            if self._vvane == "swing" and self._hvane == "swing":
+                self._swing = SWING_BOTH
+            elif self._vvane == "swing":
+                self._swing = SWING_VERTICAL
+            elif self._hvane == "swing":
+                self._swing = SWING_HORIZONTAL
+            else:
+                self._swing = SWING_OFF
+
+    async def async_will_remove_from_hass(self):
+        """Shutdown the controller when the device is being removed."""
+        await self._controller.stop()
+
+    @property
+    def icon(self):
+        """Return the icon for the current state."""
+        icon = None
+        if self._power:
+            icon = MAP_STATE_ICONS.get(self._hvac_mode)
+        return icon
+
+    def update_callback(self):
+        """Let HA know there has been an update from the controller."""
+        _LOGGER.debug("%s API sent a status update.", self._devicetype)
+        self.schedule_update_ha_state(True)
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature for the current mode of operation."""
+        return self._min_temp
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature for the current mode of operation."""
+        return self._max_temp
+
+    @property
+    def should_poll(self):
+        """Poll for updates if pyIntesisHome doesn't have a socket open."""
+        # This could be switched on controller.is_connected, but HA doesn't
+        # seem to handle dynamically changing from push to poll.
+        return True
+
+    @property
+    def hvac_modes(self):
+        """List of available operation modes."""
+        return self._hvac_modes
+
+    @property
+    def fan_mode(self):
+        """Return whether the fan is on."""
+        return self._fan_speed
+
+    @property
+    def swing_mode(self):
+        """Return current swing mode."""
+        return self._swing
+
+    @property
+    def fan_modes(self):
+        """List of available fan modes."""
+        return self._fan_modes
+
+    @property
+    def swing_modes(self):
+        """List of available swing positions."""
+        return self._swing_list
+
+    @property
+    def assumed_state(self) -> bool:
```

Comment:
Since the entity won't be available if the device isn't connected the `assumed_state` property doesn't add any value.